configs/components/puppet.rb

```diff
@@ -4,6 +4,7 @@
   pkg.build_requires "ruby-#{settings[:ruby_version]}"
   pkg.build_requires "facter"
   pkg.build_requires "hiera"
+  pkg.build_requires "rubygem-win32-dir"
```

Comment:
Another thought here is that it feels a little weird to me to require this gem, just so the Puppet installation can continue.

Of course, if we were to inline the equivalent code into `install.rb` to find the 2 paths that we care about we'd probably also need `FFI` (or could potentially use `Fiddle` which is in the box for Ruby, as long as we build / distribute it still).

Maybe we do this now, and file a ticket to clean up the tech debt? (removing `Dir::PROGRAM_FILES`, etc from `install.rb`)